### PR TITLE
Remove the deprecated local parameter for _cat/indices

### PR DESCRIPTION
**Pull request by danhermann**
The cat indices API performs several actions including an `IndicesStatsAction`. Today it accepts the `?local` parameter and passes it to the actions that support it but this parameter has no effect on the `IndicesStatsAction`. This can be surprising because `GET _cat/indices?local` looks like it might be a completely local call but in fact it still depends on every node in the cluster.

The `local` parameter was deprecated in 7.x (#62198) and this commit removes it for 8.x.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java [hunk 1]
```diff
@@ -99,10 +97,10 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
         final IndicesOptions indicesOptions = IndicesOptions.fromRequest(request, IndicesOptions.strictExpand());
-        if (request.hasParam("local")) {
-            DEPRECATION_LOGGER.deprecate("local", LOCAL_DEPRECATED_MESSAGE);
+        // needed only in v8 to catch breaking usages and may be removed in v9
+        if (request.hasParam("local")  && Version.CURRENT.major == Version.V_7_0_0.major + 1) {
+            throw new IllegalArgumentException("parameter [local] is not supported");
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java:102]: This is unnecessary, as mentioned in https://github.com/elastic/elasticsearch/pull/64867#discussion_r520703765

**Comment by danhermann** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine test this please

**Comment by danhermann** [comment]
@elasticmachine update branch

**Pull request merged**